src/nvim/window.c

```diff
@@ -7658,3 +7658,29 @@ win_T *lastwin_nofloating(void)
   }
   return res;
 }
+
+static int floating_zindex_compare(const void *a, const void *b)
+{
+  return (*(win_T **)a)->w_float_config.zindex < (*(win_T **)b)->w_float_config.zindex;
```

Comment:
better to be `(*(win_T **)b)->w_float_config.zindex - (*(win_T **)a)->w_float_config.zindex` (return value should be either negative, zero, or positive for all three possible comparisons)